### ISSUE TITLE
Message PlaceTileMessage hasn't been handled by ActionPhase phase

### ISSUE THREAD

**Issue by max-wittig** [farin/JCloisterZone#252]
While playing online the following issue occurred in version 4.0.2

### Stacktrace

```
[Thread-3] ERROR com.jcloisterzone.ui.view.GameView - Message PlaceTileMessage hasn't been handled by ActionPhase phase.
java.lang.IllegalArgumentException: Message PlaceTileMessage hasn't been handled by ActionPhase phase.
	at com.jcloisterzone.game.GameStatePhaseReducer.applyMessageOnPhase(GameStatePhaseReducer.java:143)
	at com.jcloisterzone.game.GameStatePhaseReducer.apply(GameStatePhaseReducer.java:158)
	at com.jcloisterzone.game.Game.handleInGameMessage(Game.java:244)
	at sun.reflect.GeneratedMethodAccessor22.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.jcloisterzone.wsio.MessageDispatcher.dispatchOn(MessageDispatcher.java:38)
	at com.jcloisterzone.wsio.MessageDispatcher.dispatch(MessageDispatcher.java:21)
	at com.jcloisterzone.ui.ClientMessageListener.onWebsocketMessage(ClientMessageListener.java:136)
	at com.jcloisterzone.wsio.WebSocketConnection$WebSocketClientImpl.onMessage(WebSocketConnection.java:91)
	at org.java_websocket.client.WebSocketClient.onWebsocketMessage(WebSocketClient.java:292)
	at org.java_websocket.WebSocketImpl.decodeFrames(WebSocketImpl.java:417)
	at org.java_websocket.WebSocketImpl.decode(WebSocketImpl.java:170)
	at org.java_websocket.client.WebSocketClient.run(WebSocketClient.java:229)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by farin** [comment]
It would be great if you have saved game after it happened. But I expect you probably don't have it.
Saved game contains all game events and can be replayed.

**Comment by max-wittig** [comment]
Okay I sadly don't have that. Will save it next time and update this thread, if it comes up again.

**Comment by farin** [comment]
Observation: it seems's it's caused by impropertly handled connection leak by JavaWebsocket library (at least I am able to simulate same error when placing tile/meeple during) temporaty down network.

 Going to try it it with latest JavaWebsocket (which wasn't unfortunatelly used in jcz)

**Comment by farin** [comment]
fixed by b8cfabcf735a966da37c0570aa617e018eae9668
